File: src/main/java/no/nav/bidrag/beregn/underholdskostnad/periode/UnderholdskostnadPeriode.java
package no.nav.bidrag.beregn.underholdskostnad.periode;

import java.util.List;
import no.nav.bidrag.beregn.felles.bo.Avvik;
import no.nav.bidrag.beregn.underholdskostnad.beregning.UnderholdskostnadBeregning;
import no.nav.bidrag.beregn.underholdskostnad.bo.BeregnUnderholdskostnadGrunnlag;
import no.nav.bidrag.beregn.underholdskostnad.bo.BeregnetUnderholdskostnadResultat;

public interface UnderholdskostnadPeriode {
  BeregnetUnderholdskostnadResultat beregnPerioder(BeregnUnderholdskostnadGrunnlag beregnUnderholdskostnadGrunnlag);

  List<Avvik> validerInput(BeregnUnderholdskostnadGrunnlag beregnUnderholdskostnadGrunnlag);

  static UnderholdskostnadPeriode getInstance() {
    return new UnderholdskostnadPeriodeImpl(UnderholdskostnadBeregning.getInstance());
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/underholdskostnad/periode/UnderholdskostnadPeriode.kt
package no.nav.bidrag.beregn.underholdskostnad.periode

import no.nav.bidrag.beregn.felles.bo.Avvik
import no.nav.bidrag.beregn.underholdskostnad.beregning.UnderholdskostnadBeregning
import no.nav.bidrag.beregn.underholdskostnad.bo.BeregnUnderholdskostnadGrunnlag
import no.nav.bidrag.beregn.underholdskostnad.bo.BeregnetUnderholdskostnadResultat

interface UnderholdskostnadPeriode {
    fun beregnPerioder(grunnlag: BeregnUnderholdskostnadGrunnlag): BeregnetUnderholdskostnadResultat
    fun validerInput(grunnlag: BeregnUnderholdskostnadGrunnlag): List<Avvik>

    companion object {
        fun getInstance(): UnderholdskostnadPeriode = UnderholdskostnadPeriodeImpl(UnderholdskostnadBeregning.getInstance())
    }
}
